<!DOCTYPE html>
<meta charset="utf-8">

    <head>
        <meta charset="utf-8">
        <meta http-equiv="X-UA-Compatible" content="IE=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0" />
        <meta name="description" content="">
        <meta name="author" content="Eden Zik">

        <title>Illustrating evolution of EBOV Glycoprotein</title>

        <link rel="stylesheet" href="http://fonts.googleapis.com/css?family=Lato:300,400,400italic" type="text/css">
        <link rel="stylesheet" href="http://netdna.bootstrapcdn.com/bootstrap/3.1.1/css/bootstrap.min.css" type='text/css'>
        <link rel="stylesheet" href="css/main.css" type='text/css'>
        <link rel="icon" type="image/png" href="/favicon.png" />
    </head>


<style>
body{
    width:1060px;
    margin:50px auto;
}
path {  stroke: #fff; }
path:hover {  opacity:0.9; }
rect:hover {  fill:blue; }
.axis {  font: 10px sans-serif; }
.legend tr{    border-bottom:1px solid grey; }
.legend tr:first-child{    border-top:1px solid grey; }

.axis path,
.axis line {
  fill: none;
  stroke: #000;
  shape-rendering: crispEdges;
}

.x.axis path {  display: none; }
.legend{
    margin-bottom:76px;
    display:inline-block;
    border-collapse: collapse;
    border-spacing: 0px;
}
.legend td{
    padding:4px 5px;
    vertical-align:bottom;
}
.legendFreq, .legendPerc{
    align:right;
    width:50px;
}

</style>


    <body data-spy="scroll" data-target=".navbar-example">
    
        <nav class="navbar navbar-default navbar-fixed-top" role="navigation">
          <div class="container-fluid">
            <!-- Brand and toggle get grouped for better mobile display -->
            <div class="navbar-header">
              <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#bs-example-navbar-collapse-1">
                <span class="sr-only">Toggle navigation</span>
                <span class="icon-bar"></span>
                <span class="icon-bar"></span>
                <span class="icon-bar"></span>
              </button>
              <a class="navbar-brand" href="#">Top</a>
            </div>

            <!-- Collect the nav links, forms, and other content for toggling -->
            <div class="collapse navbar-collapse" id="bs-example-navbar-collapse-1">
              <ul class="nav navbar-nav">
                <li class="active"><a href="#">Link <span class="sr-only">(current)</span></a></li>
                <li><a href="#structure">Structure</a></li>
                </li>
              </ul>
              <ul class="nav navbar-nav navbar-right">
                <iframe src="http://ghbtns.com/github-btn.html?user=edenzik&repo=178aproject&type=fork&size=large"allowtransparency="true" frameborder="0" scrolling="0"width="95" height="40"></iframe>
              </ul>
            </div><!-- /.navbar-collapse -->
          </div><!-- /.container-fluid -->
        </nav>

<div id="test" class="container">
    <h1>Illustrating evolution of EBOV Glycoprotein</h1>
    <h2>Brendan, Zevvy, Eden</h2>
    <div class="row">
        <div class="col-md-8">
          <p>We wish to create a visual web product that illustrates the geographic, chronological, genetic, and biochemical diversity of the Ebolavirus glycoprotein (EBOV GP) evolution over the past 40 years. 
          </p>
          <p>This will be accomplished by specific case studies of, but not limited to, Ebolavirus outbreaks in Yambuku, Zaire (1976); Kiwit, Zaire (1995); Sierra Leon (2014). 
          </p>
        </div>
    <div class="col-md-4">
        <div id="glmol02"></div>
    </div>
 </div>

  <div class="row">
    <div class="col-lg-8 col-md-8 col-sm-8">
      <h3>Introduction</h3>
      <p>There are five strands of ebolavirus recognized by the International Committee on Taxonomy of Viruses: Ebola virus (EBOV), Sudan virus (SUDV), Reston virus (RESTV), Ta√Ø Forest virus (TAFV), and Bundibugyo virus (BDBV). Despite the large number of outbreaks over the past 40 years, we focus our analysis on Ebola virus and outbreaks in Zaire (1976), Zaire (1995), Democractic Republic of the Congo (2007), and Sierra Leon (2014). 
      <br><br>
      The World Health Organization reports 15,351 Ebola cases with 5,459 deaths across eight countries so far in this current outbreak. Amongst the outbreak, Stephen Gire and 49 others published <i>Genomic surveillance elucidates Ebola virus origin and transmission during the 2014 outbreak</i> in Science this August and made publicly available 99 fully sequenced Ebola virus genomes from 78 patients in Sierra Leon. 
      </p>
    </div>

    <!--<div class="col-lg-6 col-md-6 col-sm-6">
      <h3>How to Use</h3>
      <p>This tool presents a variety of data about Ebola genetics. We wish to explore this topic extensively.</p>
    </div>
  </div>-->

<div id='dashboard' align="center">

</div>

    <script src="http://d3js.org/d3.v3.min.js"></script>
    <script>
        function dashboard(id, fData){
            var barColor = 'steelblue';
            function segColor(c){ return {deaths:"#ba807d", survivors:"#7db7ba"}[c]; }
            
            // compute total for each state.
            fData.forEach(function(d){d.total=d.freq.survivors+d.freq.deaths;});
            
            // function to handle histogram.
            function histoGram(fD){
                var hG={},    hGDim = {t: 60, r: 0, b: 30, l: 0};
                hGDim.w = 500 - hGDim.l - hGDim.r, 
                hGDim.h = 300 - hGDim.t - hGDim.b;
                    
                //create svg for histogram.
                var hGsvg = d3.select(id).append("svg")
                    .attr("width", hGDim.w + hGDim.l + hGDim.r)
                    .attr("height", hGDim.h + hGDim.t + hGDim.b).append("g")
                    .attr("transform", "translate(" + hGDim.l + "," + hGDim.t + ")");

                // create function for x-axis mapping.
                var x = d3.scale.ordinal().rangeRoundBands([0, hGDim.w], 0.1)
                        .domain(fD.map(function(d) { return d[0]; }));

                // Add x-axis to the histogram svg.
                hGsvg.append("g").attr("class", "x axis")
                    .attr("transform", "translate(0," + hGDim.h + ")")
                    .call(d3.svg.axis().scale(x).orient("bottom"));

                // Create function for y-axis map.
                var y = d3.scale.linear().range([hGDim.h, 0])
                        .domain([0, d3.max(fD, function(d) { return d[1]; })]);

                // Create bars for histogram to contain rectangles and freq labels.
                var bars = hGsvg.selectAll(".bar").data(fD).enter()
                        .append("g").attr("class", "bar");
                
                //create the rectangles.
                bars.append("rect")
                    .attr("x", function(d) { return x(d[0]); })
                    .attr("y", function(d) { return y(d[1]); })
                    .attr("width", x.rangeBand())
                    .attr("height", function(d) { return hGDim.h - y(d[1]); })
                    .attr('fill',barColor)
                    .on("mouseover",mouseover)// mouseover is defined below.
                    .on("mouseout",mouseout);// mouseout is defined below.
                    
                //Create the frequency labels above the rectangles.
                bars.append("text").text(function(d){ return d3.format(",")(d[1])})
                    .attr("x", function(d) { return x(d[0])+x.rangeBand()/2; })
                    .attr("y", function(d) { return y(d[1])-5; })
                    .attr("text-anchor", "middle");
                
                function mouseover(d){  // utility function to be called on mouseover.
                    // filter for selected state.
                    var st = fData.filter(function(s){ return s.State == d[0];})[0],
                        nD = d3.keys(st.freq).map(function(s){ return {type:s, freq:st.freq[s]};});
                       
                    // call update functions of pie-chart and legend.    
                    pC.update(nD);
                    leg.update(nD);
                }
                
                function mouseout(d){    // utility function to be called on mouseout.
                    // reset the pie-chart and legend.    
                    pC.update(tF);
                    leg.update(tF);
                }
                
                // create function to update the bars. This will be used by pie-chart.
                hG.update = function(nD, color){
                    // update the domain of the y-axis map to reflect change in frequencies.
                    y.domain([0, d3.max(nD, function(d) { return d[1]; })]);
                    
                    // Attach the new data to the bars.
                    var bars = hGsvg.selectAll(".bar").data(nD);
                    
                    // transition the height and color of rectangles.
                    bars.select("rect").transition().duration(500)
                        .attr("y", function(d) {return y(d[1]); })
                        .attr("height", function(d) { return hGDim.h - y(d[1]); })
                        .attr("fill", color);

                    // transition the frequency labels location and change value.
                    bars.select("text").transition().duration(500)
                        .text(function(d){ return d3.format(",")(d[1])})
                        .attr("y", function(d) {return y(d[1])-5; });            
                }        
                return hG;
            }
            
            // function to handle pieChart.
            function pieChart(pD){
                var pC ={},    pieDim ={w:250, h: 250};
                pieDim.r = Math.min(pieDim.w, pieDim.h) / 2;
                        
                // create svg for pie chart.
                var piesvg = d3.select(id).append("svg")
                    .attr("width", pieDim.w).attr("height", pieDim.h).append("g")
                    .attr("transform", "translate("+pieDim.w/2+","+pieDim.h/2+")");
                
                // create function to draw the arcs of the pie slices.
                var arc = d3.svg.arc().outerRadius(pieDim.r - 10).innerRadius(0);

                // create a function to compute the pie slice angles.
                var pie = d3.layout.pie().sort(null).value(function(d) { return d.freq; });

                // Draw the pie slices.
                piesvg.selectAll("path").data(pie(pD)).enter().append("path").attr("d", arc)
                    .each(function(d) { this._current = d; })
                    .style("fill", function(d) { return segColor(d.data.type); })
                    .on("mouseover",mouseover).on("mouseout",mouseout);

                // create function to update pie-chart. This will be used by histogram.
                pC.update = function(nD){
                    piesvg.selectAll("path").data(pie(nD)).transition().duration(500)
                        .attrTween("d", arcTween);
                }        
                // Utility function to be called on mouseover a pie slice.
                function mouseover(d){
                    // call the update function of histogram with new data.
                    hG.update(fData.map(function(v){ 
                        return [v.State,v.freq[d.data.type]];}),segColor(d.data.type));
                }
                //Utility function to be called on mouseout a pie slice.
                function mouseout(d){
                    // call the update function of histogram with all data.
                    hG.update(fData.map(function(v){
                        return [v.State,v.total];}), barColor);
                }
                // Animating the pie-slice requiring a custom function which specifies
                // how the intermediate paths should be drawn.
                function arcTween(a) {
                    var i = d3.interpolate(this._current, a);
                    this._current = i(0);
                    return function(t) { return arc(i(t));    };
                }    
                return pC;
            }
            
            // function to handle legend.
            function legend(lD){
                var leg = {};
                    
                // create table for legend.
                var legend = d3.select(id).append("table").attr('class','legend');
                
                // create one row per segment.
                var tr = legend.append("tbody").selectAll("tr").data(lD).enter().append("tr");
                    
                // create the first column for each segment.
                tr.append("td").append("svg").attr("width", '16').attr("height", '16').append("rect")
                    .attr("width", '16').attr("height", '16')
        			.attr("fill",function(d){ return segColor(d.type); });
                    
                // create the second column for each segment.
                tr.append("td").text(function(d){ return d.type;});

                // create the third column for each segment.
                tr.append("td").attr("class",'legendFreq')
                    .text(function(d){ return d3.format(",")(d.freq);});

                // create the fourth column for each segment.
                tr.append("td").attr("class",'legendPerc')
                    .text(function(d){ return getLegend(d,lD);});

                // Utility function to be used to update the legend.
                leg.update = function(nD){
                    // update the data attached to the row elements.
                    var l = legend.select("tbody").selectAll("tr").data(nD);

                    // update the frequencies.
                    l.select(".legendFreq").text(function(d){ return d3.format(",")(d.freq);});

                    // update the percentage column.
                    l.select(".legendPerc").text(function(d){ return getLegend(d,nD);});        
                }
                
                function getLegend(d,aD){ // Utility function to compute percentage.
                    return d3.format("%")(d.freq/d3.sum(aD.map(function(v){ return v.freq; })));
                }

                return leg;
            }
            
            // calculate total frequency by segment for all state.
            var tF = ['deaths','survivors'].map(function(d){ 
                return {type:d, freq: d3.sum(fData.map(function(t){ return t.freq[d];}))}; 
            });    
            
            // calculate total frequency by state for all segment.
            var sF = fData.map(function(d){return [d.State,d.total];});

            var hG = histoGram(sF), // create the histogram.
                pC = pieChart(tF), // create the pie-chart.
                leg= legend(tF);  // create the legend.
        }
    </script>

    <script>
        var freqData=[
        {State:'Zaire(1976)',freq:{deaths:280, survivors:38}}
        ,{State:'Zaire(1995)',freq:{deaths:254, survivors:61}}
        ,{State:'D.R.Congo(2007)',freq:{deaths:187, survivors:77}}
        ,{State:'S.L.(2014)',freq:{deaths:1267, survivors:4923}}
        ];

        dashboard('#dashboard',freqData);
    </script>
    </div>

    <p align="center">Total cases and deaths corresponding to four major outbreaks of Ebola virus. From left to right: Zaire, 1976, 318 reported cases with 88% <br>fatality; Zaire, 1995, 315 reported cases with 81% fataility; Democratic Republic of Congo, 2007, 264 reported cases with 71% fatality;<br> Sierra Leon, 2014, 6190 reported cases with 20% fatality.</p>

    <div class="row">
    <p>We perform an exploratory analysis in attempt to find functional consequences to the changes present in the current strand of Ebola virus.<b>(More definite conclusions here)</b></p>
    </div>

    <div class="row">
    <h3>What does the Ebola virus genome code for?</h3>
    Ebola virus is made up of a linear sequnce of ~18,959 basepairs which codes for 7 proteins: 
    (Insert the info currently on Eden's page)
    <!-- Figure of ebolavirus from google docu and figure from Gire paper --> 
    <!-- Table of amino acids, coding regions, and their biological function -->
    <!-- Overview of how Ebola virus proteins work-->

    <!-- Show overall change, still don't know how to do this....just make a table? I maybe better to show a rooted tree? -->
    <!-- Show mutations/bp-year -->

    <!-- Show chart of how the amino acids have mutated over time -->  

    <!-- Conserved regions, biological significance -->
    
    </div>

   <div class="row">  
            <h3>How has the virus changed over time?</h3>
            <p>Gire 2014 found that a rooted-to-tip distance correlated better with rooting from the 1976 outbreak than the 2014 outbreak. We utilized Clustal Omega to find the sequence identity between each outbreak and discovered that, the outbreak has become increasingly divergent from the 1976 strand over time.</p>

            <br>

            <img src="img/identity.png" alt="Ebola Virus" class="img-rounded" width="550" align="middle">

            <br>

            (Graph of sequence diveregence)

            <br><br>

            <p>Presently, the 2007 strand is only 97.02% similar to the 1976 outbreak. While that may not seem much, this difference corresponds to just over 560 nucleotides in difference.</p>

            Using sequence divergence, we reproduced a calculation made by Gire 2014 in which they claimed there have been 8e-04 mutations/bp-year (mutations per basepair per year) in the Ebola virus genome. We replicated this calculation and performed similar analysis between each outbreak. Interestingly, it seems that the virus has mutated more quickly relative to other outbreaks by about 30%.

            <br><br>

            (Graph for mutations/bp-year)
    </div>

    <div class="row">
        <h3>Where do these changes occur?</h3>
        <p>(Insert section from Eden's version)</p>
    </div>

    <div class="row">
        <h3>Conservation within the GP coding region</h3>
        <p>The relatively large number of mutations within the glycoprotein coding region indictates that there may be some importance to biological function worth investigating, so shift our focus from nucelotides to amino acids. 
        <br><br>
        We proceeded by searching for non-conserved domains within the 2014 outbreak glycoprotein protein structure using a tool called Pfam and comparing to the conserved domain database. We inputted the protein sequence from the 2014 outbreak to Pfam. 

        <!--<div class="col-md-12 diagram">
        <img src="img/matrix.png" alt="Ebola Virus" class="img-rounded">
        </div>-->
        <br><br>
        <img src="img/matrix.png" alt="Ebola Virus" class="img-rounded" width="550" align="middle">
        <br>(Probably will be resized?)
        <br><br>
        <h4>Conserved Regions</h4>
            <p>Our results show that there are two sections which are conserved and this image shows conserved domains from amino acids 557 to 629.<p>
            <br>
            <p> This image shows conserved domains from amino acid 557 to 629. Immediately appearent is the commonalities with other viruses, including HIV, such as the immunosuppressive region. The HR domains are conserved within different Ebola virus species. HR domains are responsible for for fusion with the host cell membrane - when folded, they form the core of the glycoprotein complex and, when activated, extend to grab onto host cell membranes. <br>

            <img src="img/HRdomain.png" alt="Ebola Virus" class="img-rounded" width="550" align="middle">
            <br>
            (Future Virol. 2009; 4(6): 621‚Äì635.).

            <br><br>

            (animated images?)

            <br><br>

            The other conserved region is the glycan protected region which give the virus protection from antibodies. The first 250 amino acids of this region have seen no change from 1976 to 2014. This domain is protected in vitro by glycans, and seems to mediate binding to the host cell via a receptor, but the exact mechanism is still unknown. 

            <img src="img/glycan.png" alt="Ebola Virus" class="img-rounded" width="550" align="middle"></p>
        <h4>Nonconserved regions</h4>
        <p>The areas of the glycoprotein which have changed are the small regions exposed to binding by antibodies.  Below is the crystal structure of Ebola GP bound to antibodies from a human survivor from the 1995 Kitwit outbreak. The yellow is the GP core, while the green are antibodies.</p>

        <br><br>
        (Zevvy animation - antibodies .pdb)

        <br><br>
        <p>The areas of the glycoprotein which have changed are the small regions exposed to binding by antibodies.  Below is the crystal structure of Ebola GP bound to antibodies from a human survivor from the 1995 Kitwit outbreak. The yellow is the GP core, while the green are antibodies.</p>

        <h4>Significant Changes</h4>
        (Eden's Beautiful Table) <br>

        <p>Gire 2014 identified a number of significant protein changes across sequenced patients in Sierra Leon, illustrated in the table. The conserved region of the first 250 amino acids is observed in their findings as well, as the frequency of mutation is not shown.</p><br>

        <p>Although we know now know how the Ebola virus has changed, we do not know the corresponding biological function to these mutations. Our discovery of non-conserved regions being the small regions of the Glycoprotein indictate a potential searching point for biomedical research. This is particularly alarming as a vaccine may be rendered ineffective quicker than expected - especially with such intrahost and interhost variation present in the current outbreak.</p>

        <h4>Consensus sequence</h4>
        We wrote a MATLAB script to generate a consensus sequence from clustal omega multiple sequence alignments to generate a consensus sequence of the Ebola virus across our studied outbreaks. The script is currently written with the assumption that alignment was performed on four sequences; however, it may be easily generalized to generate consensus sequences for a different number. 
        <br><br>
        You download our consensus nucleotide sequence, 18960 nucleotides long, here: <a href="consensus.txt">consensus sequence</a>.
        <br>
        And our MATLAB script, here: <a href="consensus.m">consensus.m</a>.
    </div>


</body>
</html>
